JSTests/stress/math-hypot.js

```diff
@@ -0,0 +1,180 @@
+    shouldBe(Math.hypot(-1e308, -1e308, -1e308), 1.7320508075688772e308);
+    shouldBe(Math.hypot(1e308, 1e308, 1e308, 1e308), Infinity);
+    shouldBe(Math.hypot(-1e308, -1e308, -1e308, -1e308), Infinity);
+    shouldBe(Math.hypot(1e308, 1e308, 1e308, 1e308, -1e308), Infinity);
+    shouldBe(Math.hypot(0.1), 0.1);
+    shouldBe(Math.hypot(0.1, 0.1), 0.14142135623730953);
+    shouldBe(Math.hypot(0.1, -0.1), 0.14142135623730953);
+    shouldBe(Math.hypot(1e308, 1e308, 0.1, 0.1, 1e30, 0.1, -1e30, -1e308, -1e308), Infinity);
+    shouldBe(Math.hypot(1e30, 0.1, -1e30), 1.4142135623730953e30);
+    shouldBe(Math.hypot(1e30, 1000.11, -1e30), 1.4142135623730953e30);
+    shouldBe(Math.hypot(1e30, Number.MAX_SAFE_INTEGER, -1e30), 1.4142135623730953e30);
+
+    // big and small numbers
+    shouldBe(Math.hypot(...Array(100).fill(1)), 10);
+    shouldBe(Math.hypot(...Array(10000).fill(1)), 100);
+    shouldBe(Math.hypot(8.98846567431158e307, 8.988465674311579e307, -1.7976931348623157e308), Infinity);
+    shouldBe(Math.hypot(-5.630637621603525e255, 9.565271205476345e307, 2.9937604643020797e292), 9.565271205476345e307);
+    shouldBe(
+        Math.hypot(
+            6.739986666787661e66,
+            2,
+            -1.2689709186578243e-116,
+            1.7046015739467354e308,
+            -9.979201547673601e291,
+            6.160926733208294e307,
+            -3.179557053031852e234,
+            -7.027282978772846e307,
+            -0.7500000000000001
+        ),
+        Infinity
+    );
+    shouldBe(
+        Math.hypot(
+            0.31150493246968836,
+            -8.988465674311582e307,
+            1.8315037361673755e-270,
+            -15.999999999999996,
+            2.9999999999999996,
+            7.345200721499384e164,
+            -2.033582473639399,
+            -8.98846567431158e307,
+            -3.5737295155405993e292,
+            4.13894772383715e-124,
+            -3.6111186457260667e-35,
+            2.387234887098013e180,
+            7.645295562778372e-298,
+            3.395189016861822e-103,
+            -2.6331611115768973e-149
+        ),
+        1.2711610061536464e308
+    );
+    shouldBe(
+        Math.hypot(
+            -1.1442589134409902e308,
+            9.593842098384855e138,
+            4.494232837155791e307,
+            -1.3482698511467367e308,
+            4.494232837155792e307
+        ),
+        Infinity
+    );
+    shouldBe(
+        Math.hypot(-1.1442589134409902e308, 4.494232837155791e307, -1.3482698511467367e308, 4.494232837155792e307),
+        Infinity
+    );
+    shouldBe(
+        Math.hypot(9.593842098384855e138, -6.948356297254111e307, -1.3482698511467367e308, 4.494232837155792e307),
+        1.5819637896591838e308
+    );
+    shouldBe(Math.hypot(-2.534858246857893e115, 8.988465674311579e307, 8.98846567431158e307), 1.2711610061536462e308);
+    shouldBe(Math.hypot(1.3588124894186193e308, 1.4803986201152006e223, 6.741349255733684e307), 1.5168484694516577e308);
```

Comment:
> [!NOTE]  
> The test results are consistent with Chrome and Firefox, except for the case`shouldBe(Math.hypot(1.3588124894186193e308, 1.4803986201152006e223, 6.741349255733684e307), 1.5168484694516577e308);`.
> 
> In WebKit, it returns 1.5168484694516577e308, while Chrome and Firefox return 1.5168484694516575e308.
> 
> This PR does not change the result of that calculation since it’s a 3-argument case.
> The WebKit result is actually consistent with the behavior of builtin `hypot` functions in Swift, C++, Rust, and Python.